fable-test-app/src/pages/SubmitHoliday.tsx

```diff
@@ -1,16 +1,111 @@
-import React from 'react';
+import React, { useState, useRef } from 'react';
 
 // Components (internal)
-import { DateModified, Heading, Text } from '../components';
+import {
+  DateModified,
+  Heading,
+  Text,
+  StepOne,
+  StepTwo,
+  Success
+} from '../components';
 
 const SubmitHoliday: React.FC = () => {
+  const [step, setStep] = useState(1);
+  const [formData, setFormData] = useState({
+    holidayname: '',
+    newholiday: '',
+    holidaydate: '',
+    learnofholiday: '',
+    holidaytype: {
+      federal: false,
+      national: false,
+      other: false
+    },
+    otherholiday: '',
+    province: '',
+    image: null,
+    fullname: '',
+    email: ''
+  });
+  const [focusHeading, setFocusHeading] = useState(false);
+
+  const heading = useRef<HTMLGcdsStepperElement>(null);
+
+  // Handle form inputs to set into state
+  const handleInputChange = (e: any) => {
+    const { name, value } = e.target;
+    if (name === "newholiday") {
+      setFormData({...formData, [name]: e.detail});
+    } else if (name === "holidaytype") {
+      let type = formData.holidaytype;
+      setFormData({...formData, [name]: { ...type, [value]: !type[value as keyof typeof type] }});
+    } else {
+      setFormData({...formData, [name]: value});
+    }
+  };
+
+  const previousStep = () => {
+    setFocusHeading(true);
+    setStep(step - 1);
+    setTimeout(() => {
+      heading.current!.focus();
+      setFocusHeading(false);
+    }, 50);
+  };
+
+  // Errors array to check if all fields validated properly
+  let errors: (EventTarget | null)[] = [];
+
+  // Error handling
+  document.addEventListener('gcdsError', (e) => {
+    if (!errors.includes(e.target)) {
+      errors.push(e.target);
+    }
+  })
+  document.addEventListener('gcdsValid', (e) => {
+    if (errors.includes(e.target)) {
+      errors.splice(errors.indexOf(e.target), 1)
+    }
+  })
+
   return (
     <section>
       <Heading tag="h1">Submit a holiday</Heading>
       <Text>
         This is a form you can use to submit a holiday that we're missing. There's a few steps involved to showcase our form components.
       </Text>
 
+      <form
+        onSubmit={(e) => {
+          e.preventDefault();
+          setTimeout(() => {
+            if (errors.length == 0) {
+              setStep(step + 1);
+            }
+          }, 50);
+        }}
+      >
+        {step < 2 ?
```

Comment:
Could we do this instead?
```suggestion
        {step === 1 ?
```